### PR TITLE
Exe 1941 pixelatorR tasks

### PR DESCRIPTION
**Pull request by ludvigla**
## Description

This PR adds a number of tasks to simplify package development. Below is a list of the tasks included:

- `setup-env-bare` : Create a new environment with micromamba with base R installed
- `setup-env-pixelatorR` : Create a new environment with micromamba with pixelatorR and its dependencies installed
- `install-pixelatorR-with-pak` : Install pixelatorR and its dependencies to an existing micromamba environment using pak. This can be useful as an alternative option to facilitate install on arm architectures.
- `document` : Runs devtools document on the R package.
- `test-all` : Runs all tests using testthat.
- `test-examples` : Runs all examples using devtools::test()
- `test-staged-files` : Runs tests for staged test R scripts. This can be useful to set up pre-commit hooks.
- `style-staged-files` : Runs code styling for staged R scripts. This can be useful to set up pre-commit hooks.

Fixes: exe-1941

## How Has This Been Tested?

These additions do not affect the package itself and therefore this PR does not include any new tests.

## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked my code and documentation and corrected any misspellings.
- [x] I have run R CMD check on the package and it passes without errors or warnings (notes can be acceptable if motivated)
- [ ] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+      The environment name can be set using the ENV_NAME environment variable.
+
+      Examples:
+      task setup-env-pixelatorR ENV_NAME=my_env
+      task setup-env-pixelatorR ENV_NAME=my_env MICROMAMBA_ARGS="--platform osx-64"
+
+    vars:
+      MICROMAMBA_ARGS: '{{ .MICROMAMBA_ARGS | default "" }}'
+      ENV_NAME: '{{.ENV_NAME | default "r-pixelator"}}'
```
Comment (maxkarlsson) [inline Taskfile.yml:97]: Could be good to mention the default if you add documentation to developer docs

File: inst/dev/install-deps.R [hunk 2]
```diff
@@ -0,0 +1,26 @@
+options(repos = c(CRAN = "https://cloud.r-project.org"))
+if (getRversion() < "4.4.1") {
+  cli_alert_error("Please upgrade to R 4.4.1 or later.")
```
Comment (maxkarlsson) [inline inst/dev/install-deps.R:3]:
```suggestion
  cli::cli_alert_error("Please upgrade to R 4.4.1 or later.")
```

Comment (maxkarlsson) [reply inst/dev/install-deps.R:3]: I was getting an error when running task install-with-pak that cli_alert_error can't be found.

**Review by maxkarlsson — APPROVED** [review]
Looks real good! Great addition for us, so thank you for doing it!

The only thing I am missing is some documentation in the developers.md file like we discussed. I think it is enough if we just list available tasks and briefly how to run it there. Perhaps a sentence about when certain tasks should be run.

**Review by vincent-van-hoef — COMMENTED** [review]
All task commands worked except for `install-with-pak` 

```
Error in cli_alert_error("Please upgrade to R 4.4.1 or later.") : 
  could not find function "cli_alert_error"
Execution halted
task: Failed to run task "install-with-pak": exit status 1
```

and `setup-env-pixelatorR`

```
error    libmamba Could not solve for environment specs
    The following packages are incompatible
    ├─ bioconductor-limma does not exist (perhaps a typo or a missing channel);
    ├─ r-duckdb does not exist (perhaps a typo or a missing channel);
    ├─ r-ggraph does not exist (perhaps a typo or a missing channel);
    ├─ r-graphlayouts does not exist (perhaps a typo or a missing channel);
    ├─ r-harmony does not exist (perhaps a typo or a missing channel);
    └─ r-hdf5r does not exist (perhaps a typo or a missing channel).
critical libmamba Could not solve for environment specs
Failed to create environment 'r-pixelator'.
task: Failed to run task "setup-env-pixelatorR": exit status 1
```

mamba used the arm64 channels for this, but perhaps it is not suitable for this? 

Otherwise great! Let me know if you want me tot test some more!

File: Taskfile.yml (hunk 1 above)
Comment (ludvigla) [reply Taskfile.yml:97]: Good suggestion! I will add that

File: inst/dev/install-deps.R (hunk 2 above)
Comment (ludvigla) [reply inst/dev/install-deps.R]: Perfect, thanks!

**Comment by ludvigla** [comment]
> All task commands worked except for `install-with-pak`
> 
> ```
> Error in cli_alert_error("Please upgrade to R 4.4.1 or later.") : 
>   could not find function "cli_alert_error"
> Execution halted
> task: Failed to run task "install-with-pak": exit status 1
> ```
> 
This should be addressed now, thanks!

> and `setup-env-pixelatorR`
> 
> ```
> error    libmamba Could not solve for environment specs
>     The following packages are incompatible
>     ├─ bioconductor-limma does not exist (perhaps a typo or a missing channel);
>     ├─ r-duckdb does not exist (perhaps a typo or a missing channel);
>     ├─ r-ggraph does not exist (perhaps a typo or a missing channel);
>     ├─ r-graphlayouts does not exist (perhaps a typo or a missing channel);
>     ├─ r-harmony does not exist (perhaps a typo or a missing channel);
>     └─ r-hdf5r does not exist (perhaps a typo or a missing channel).
> critical libmamba Could not solve for environment specs
> Failed to create environment 'r-pixelator'.
> task: Failed to run task "setup-env-pixelatorR": exit status 1
> ```
> 
> mamba used the arm64 channels for this, but perhaps it is not suitable for this?
> 
> Otherwise great! Let me know if you want me tot test some more!

This is expected on arm because those dependencies are missing. This is the reason why the pak alternative exists. I have added some documentation in `DEVELOPERS.md` which hopefully explains this. Basically, the `task setup-env-pixelatorR` should work on all systems except Max OS on arm unless you use an emulation with Rosetta 2. Otherwise, you need to use `task setup-env-bare` and then `task install-pixelatorR-with-pak`.

If you could have a look at `DEVELOPERS.md` that would be fantastic!

**Comment by ludvigla** [comment]
> Looks real good! Great addition for us, so thank you for doing it!
> 
> The only thing I am missing is some documentation in the developers.md file like we discussed. I think it is enough if we just list available tasks and briefly how to run it there. Perhaps a sentence about when certain tasks should be run.

I have added instruction in `DEVELOPERS.md`. Let me know if you think we should go a bit more in depth.

**Comment by maxkarlsson** [comment]
> > Looks real good! Great addition for us, so thank you for doing it!
> > The only thing I am missing is some documentation in the developers.md file like we discussed. I think it is enough if we just list available tasks and briefly how to run it there. Perhaps a sentence about when certain tasks should be run.
> 
> I have added instruction in `DEVELOPERS.md`. Let me know if you think we should go a bit more in depth.

Looks great! I edited a few typos.

**Pull request merged**